sdk/Pulumi/Resources/ComponentResource.cs

```diff
@@ -61,7 +62,38 @@ public ComponentResource(string type, string name, ComponentResourceOptions? opt
         /// state as quickly as possible (instead of waiting until the entire application completes).
         /// </summary>
         protected void RegisterOutputs()
-            => RegisterOutputs(ImmutableDictionary<string, object?>.Empty);
+        {
+            var outputs = new Dictionary<string, object?>();
+            var currentType = GetType();
+            foreach (var prop in currentType.GetProperties())
```

Comment:
GetProperties only returns public props right?

Reply:
Yes, the unit test now shows that a private property isn't registered